integration-tests/deployment/changeset.go

```diff
@@ -1,12 +1,31 @@
 package deployment
 
 import (
+	"encoding/json"
+	"errors"
+
 	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
 )
 
-// Services as input to CI/Async tasks
+// Changeset represents a set of changes to be made to an environment.
+// The configuration contains environment specific inputs for a specific changeset.
+// The configuration might contain for example the chainSelectors to apply the change to
+// or existing environment specific contract addresses.
+// Its recommended that changesets operate on a small number of chains (e.g. 1-3)
+// to reduce the risk of partial failures.
+type Changeset func(e Environment, config interface{}) (ChangesetOutput, error)
```

Comment:
Having an explicit Changeset (and view state) notion makes sense to me.

I'm trying to roll around the idea that it's a function-type, rather than an interface. You can pull in an interface, but I don't see where we're forcing correct types, except in the unused var, which looks like a convention we could forget, rather than a type we have to conform to to even use the system. Or is there something I'm missing here? Is there a place where we consume (formally, so the compiler can enforce it) the ViewState and Changeset functions?

I guess this is fine in the sense that there isn't necessarily any reason for a struct with any particular data, if these are pure functions anyway, but is there any functionality that attaches to a change set, any metadata about the changeset, that we envision?

I guess it's not terribly hard to fix later, I'm just trying to get a sense of where we want it to go.

Reply:
Maybe some metadata it would be useful would be the expected shape/content of the configuration and even some description of what each configuration setting does?.

Should we split the config in two pieces?. Product specific config and Environment specific config?. I guess the environment config shape would be common to all products and we can even reuse instances of the environment config. Then we can use the shape of each config to do static validation of the input configs.

Reply:
> I guess this is fine in the sense that there isn't necessarily any reason for a struct with any particular data, if these are pure functions anyway, but is there any functionality that attaches to a change set, any metadata about the changeset, that we envision?

yeah originally was thinking about a struct with data, but couldn't think about any metadata. Most of the metadata (name ordering etc) is applied in chainlink-deployments. 

> Product specific config and Environment specific config?. I guess the environment config shape would be common to all products

I can't see env config being truly universal, for example maybe the changeset only proposes new jobs and doesn't touch any chains. Maybe once we see more use cases we can apply more structure